src/diagram/Diagram.ts

```diff
@@ -0,0 +1,151 @@
+import { Config } from "dompurify";
+import { Field } from "./Field";
+import { Configuration } from "../config/Configuration";
+import { BooleanOption, EnumOption, RangeOption } from "../config/Option";
+
+interface FieldPair {
+  name: string;
```

Comment:
Use `readonly`? cuz it is for class `Memeto` and I want to make sure the values in the Memento class are unmodifiable.

Reply:
updated.